### PR TITLE
feat(chat): Implement authentication flow with login UI and session management

### PR DESCRIPTION
**Pull request by ccastrotrejo**
## Commit Type

- [x] feature - New functionality
- [ ] fix - Bug fix
- [ ] refactor - Code restructuring without behavior change
- [ ] perf - Performance improvement
- [ ] docs - Documentation update
- [ ] test - Test-related changes
- [ ] chore - Maintenance/tooling

## Risk Level

- [ ] Low - Minor changes, limited scope
- [x] Medium - Moderate changes, some user impact
- [ ] High - Major changes, significant user/system impact

## What & Why

This PR implements a comprehensive authentication flow for the iframe chat application. It adds:

1. **Login UI Component**: A new `LoginPrompt` component with Fluent UI styling that displays when users need to authenticate
2. **Enhanced Auth Handler**: Improved authentication handling with popup-based login flow that monitors auth status
3. **Component Reorganization**: Refactored multi-session chat components and session list into organized subdirectories with separated styles
4. **Theme Integration**: Wrapped components in FluentProvider for consistent theming across light/dark modes

The changes improve the user experience by providing a clear, branded login interface instead of abrupt redirects when authentication is required.

## Impact of Change

- **Users**: Users will now see a professional login prompt when authentication is required, with better visual feedback during the login process. The authentication flow is more seamless with popup-based login instead of full page redirects.
- **Developers**: 
  - New `LoginPrompt` component available for reuse
  - Updated `authHandler` API with `onLoginRequired` callback replacing `onLogoutComplete`
  - Reorganized component structure in `apps/iframe-app/src/components/` with subdirectories for `MultiSessionChat` and `SessionList`
  - Enhanced authentication utilities including `openLoginPopup` and `checkAuthStatus` functions
- **System**: 
  - Better authentication state management with polling-based status checks
  - Improved error handling for authentication failures
  - FluentProvider wrapping ensures consistent theme application

## Test Plan

- [x] Unit tests added/updated
- [ ] E2E tests added/updated
- [x] Manual testing completed
- [x] Tested in: Local development environment with mock authentication scenarios

**Tests Updated**:
- Updated `authHandler.test.ts` with new test cases for `openLoginPopup` and `checkAuthStatus`
- Updated `config-parser.test.ts` with comprehensive tests for `getAgentBaseUrl` utility function
- Tests cover popup success/failure scenarios, cross-origin handling, and URL parsing edge cases

**Manual Testing**:
- Verified login prompt displays correctly in both light and dark themes
- Tested authentication flow with popup opening, user login, and successful return
- Verified component reorganization doesn't break existing functionality
- Confirmed FluentProvider theming works across all chat components

## Contributors

@ccastrotrejo

## Screenshots/Videos

https://github.com/user-attachments/assets/201d990c-709a-4690-8c34-29b034a235d3

### CODE REVIEW

File: apps/iframe-app/src/components/LoginPrompt/LoginPrompt.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export function LoginPrompt({ onLogin, isLoading = false }: LoginPromptProps) {
+  const styles = useLoginPromptStyles();
+
+  return (
+    <div className={styles.container}>
+      <Card className={styles.card}>
+        <div className={styles.iconContainer}>
+          <PersonRegular className={styles.icon} />
+        </div>
+        <Title3>Sign in required</Title3>
```
Comment (rllyy97) [inline apps/iframe-app/src/components/LoginPrompt/LoginPrompt.tsx:19]: Do we need INTL here? (And the other files, I'm not familiar with this package yet)

**Review by rllyy97 — APPROVED** [review]

File: apps/iframe-app/src/components/LoginPrompt/LoginPrompt.tsx (hunk 1 above)
Comment (ccastrotrejo) [reply apps/iframe-app/src/components/LoginPrompt/LoginPrompt.tsx:19]: I am not quite sure yet. I think eventually we will
